CIAMP-1505: add documentation for MFA (#9372)
circleci/circleci-docs: jekyll/_cci2/authentication/mfa.adoc

---
contentTags:
  platform:
  - Cloud
---
= Multi-factor authentication (MFA)
:page-layout: classic-docs
:page-liquid:
:page-description: MFA Overview
:icons: font
:experimental:

> NOTE: MFA for CircleCI is available for users who access their accounts using email and password. Users who use social login methods to access their accounts (such as GitHub or Bitbucket), may use their login provider's MFA offering. MFA is currently optional. Starting December 1, 2025 MFA will be required for all users who log in with email and password.

Reviewer: ```suggestion
NOTE:  MFA is currently optional. Starting December 1, 2025 MFA will be required if you log in to CircleCI with email and password.

Multi-factor authentication (MFA) for CircleCI is available to you if you sign in to your CircleCI account using email and password. If you use a social login method to access your CircleCI account (such as GitHub or Bitbucket), may you can use your login provider's MFA offering.
```

[#introduction]
== Introduction

> Multi-factor authentication (MFA) is an additional layer of security for your CircleCI account. We strongly recommend that you set up and enable it.

Reviewer: ```suggestion
MFA is an additional layer of security for your CircleCI account. We strongly recommend that you set up and enable it.
```

With MFA enabled, an additional verification step is required to access your account and make changes to account authentication settings, such as email or password.
This means that even if your account password is compromised, only someone with access to the additional verification factor can

> access the account. The additional verification step takes the form of providing a One-Time Password (OTP).

Reviewer: ```suggestion
access the account. The additional verification step takes the form of providing a one-time password (OTP).
```

[#setup-mfa]

> == Setup MFA

Reviewer: ```suggestion
== Set up MFA
```

To configure MFA on your account, follow these steps:

. In the link:https://app.circleci.com/home/[CircleCI web app], select your profile from the upper right corner, then select **User Settings**.

> . Select **Password & authentication** from the sidebar.

Reviewer: ```suggestion
. Select **Password** from the sidebar.
```

Reviewer: Currently it is just labeled Password, is this set to change?

Author: Yes, we plan to change this soon: https://github.com/circleci/circleci-docs/pull/9372#discussion_r2078249070

the application whenever this is requested in the CircleCI web app. The OTP will be displayed in the authenticator app, and you must input this code when prompted in the CircleCI web app.

[#mfa-recovery-codes]
== MFA recovery codes

WARNING: If you have MFA enabled, and lose access to your MFA factor and recovery code, then you will lose access to your account. You must ensure that you do not lose the recovery code.

If you lose access to your MFA factor (for example, by losing access to the authenticator application), you may use the recovery code as a second factor instead. A recovery code may only be used once, after which a new recovery code is generated and shared with you in the CircleCI web app. Whenever a new recovery code is generated, it is important to always save it somewhere safe.

It is also possible to intentionally regenerate your MFA recovery code. To do this, follow these steps:

. In the link:https://app.circleci.com/home/[CircleCI web app], select your profile from the upper right corner, then select **User Settings**.

> . Select **Password & authentication** from the sidebar.

Reviewer: ```suggestion
. Select **Password** from the sidebar.
```

Author: Same here: https://github.com/circleci/circleci-docs/pull/9372#discussion_r2078249070

. On the Password & authentication page, in the **Multi-factor authentication**section, select **Add/edit authenticator app**.
. Input your password at the password prompt.
. Input your OTP at the OTP prompt. Once these have been successfully submitted, your OTP ID and recovery code ID will be displayed.
. Select the button to **Regenerate recovery code** on the recovery code row.
. At the prompt to regenerate the recovery code, select `Yes, regenerate code`. This will generate a new code and invalidate the previous recovery code. Store the new code somewhere safe. It will not be displayed again.

image::../../img/docs/authentication/mfa-regenerate-recovery-code.png[Password and authentication page with regenerate recovery code button]

> image::../../img/docs/authentication/mfa-display-recovery-code.png[Password and authentication page displaying new recovery code]

Reviewer: @emilywoods I'm wondering if we can come up with a few troubleshooting/FAQs for this page?

Author: Yes, for sure! Would it be ok for me to add this in a follow-up pr?

---
contentTags:
  platform:
  - Cloud
---
= Multi-factor authentication (MFA)
:page-layout: classic-docs
:page-liquid:
:page-description: MFA Overview
:icons: font
:experimental:

> NOTE: MFA for CircleCI is available for users who access their accounts using email and password. Users who use social login methods to access their accounts (such as GitHub or Bitbucket), may use their login provider's MFA offering.

Author: Added here: https://github.com/circleci/circleci-docs/pull/9372/commits/f385599ce94cde63f1dd3e1471c7873daebbebdb

> Multi-factor authentication (MFA) for CircleCI is available to you if you sign in to your CircleCI account using email and password. If you use a social login method to access your CircleCI account (such as GitHub or Bitbucket), may you can use your login provider's MFA offering.

Reviewer: ```suggestion
Multi-factor authentication (MFA) for CircleCI is available if you sign in to your CircleCI account using email and password. If you use a social login method to access your CircleCI account (such as GitHub or Bitbucket), you can use your login provider's MFA offering.
```

== Introduction

Multi-factor authentication (MFA) is an additional layer of security for your CircleCI account. We strongly recommend that you set up and enable it.
With MFA enabled, an additional verification step is required to access your account and make changes to account authentication settings, such as email or password.
This means that even if your account password is compromised, only someone with access to the additional verification factor can
access the account. The additional verification step takes the form of providing a One-Time Password (OTP).

[#setup-mfa]
== Setup MFA

To configure MFA on your account, follow these steps:

. In the link:https://app.circleci.com/home/[CircleCI web app], select your profile from the upper right corner, then select **User Settings**.

> . Select **Password** from the sidebar.

Reviewer 2: ~ can we make this say Select `Password & authentication`. That's what design wants the nav link to be called, I just haven't updated it yet (didn't want to do it until we actually enabled MFA and had something other than just passwords on that page)

Author: aha, so both the title in the sidebar and on the page itself will have the text `Password & authentication`? good catch then, i'll fix this.

. You may close the window to complete the setup. When MFA is successfully configured, it is marked as **Enabled** on the Password & authentication page.

image::../../img/docs/authentication/mfa-enabled.png[Password and authentication page with MFA enabled]

[#using-mfa]
== Using MFA on your account

Once you have MFA enabled, then you must provide a second verification factor whenever you log into your account.

image::../../img/docs/authentication/mfa-otp-login.png[Login page with request for OTP]

You will also be asked to provide a second factor whenever you attempt to change the authentication settings of your account (for example updating your email or password).

Currently CircleCI supports MFA using authenticator applications only. Once you have configured MFA, as per the previous section, then you must use the OTP generated by
the application whenever this is requested in the CircleCI web app. The OTP will be displayed in the app, and you must input this code when prompted in the CircleCI web app.

> You may disable MFA in the **Password and authentication** page by deleting your OTP factor. If you do this, then we recommend that you re-enable MFA at your earliest convenience for the security benefit it provides.

Reviewer 2: @emilywoods @labsquared 🤔 I wonder if we want to actually advertise this considering we'll be removing the functionality later?

Author: Yeah, I was a bit on the fence about whether or not to include it, for the same reason. Happy to remove this, unless you have other thoughts @labsquared?

Author: I have removed this paragraph, given that we'll be removing this functionality later and I don't think the paragraph added much value

[#mfa-recovery-codes]
== MFA recovery codes

> WARNING: If you have MFA enabled, and lose access to your MFA factor and recovery code, then you will lose access to your account. You must ensure that you do not lose the recovery code.

Reviewer 3: ~~You will have to reach out to support to help unblock you

> WARNING: If you have MFA enabled, and lose access to your MFA factor and recovery code, then you will lose access to your account. You must ensure that you do not lose the recovery code.

Reviewer 3: ~~Should we allow support to be able to do recoveries? It would require identifying that the user is who they say they are
